test/integration/extensions/controller/backupbucket_test.go

```diff
@@ -0,0 +1,298 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controller_test
+
+import (
+	"context"
+	"time"
+
+	backupbucketcontroller "github.com/gardener/gardener/extensions/pkg/controller/backupbucket"
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	"github.com/gardener/gardener/pkg/operation/common"
+	"github.com/gardener/gardener/pkg/utils"
+	"github.com/gardener/gardener/test/framework"
+	"github.com/gardener/gardener/test/integration/extensions/controller/fake"
+	"github.com/gardener/gardener/test/integration/extensions/controller/fake/backupbucket"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	gomegatypes "github.com/onsi/gomega/types"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/util/retry"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+var _ = Describe("BackupBucket", func() {
+	AfterEach(func() {
+		framework.RunCleanupActions()
+	})
+
+	It("should successfully create and delete a BackupBucket (ignoring operation annotation)", func() {
+		prepareAndRunTest(true)
+	})
+
+	It("should successfully create and delete a BackupBucket (respecting operation annotation)", func() {
+		prepareAndRunTest(false)
+	})
+})
+
+func prepareAndRunTest(ignoreOperationAnnotation bool) {
+	By("setup and start manager")
+	mgr, err := createAndStartManager(ignoreOperationAnnotation)
+	Expect(err).NotTo(HaveOccurred())
+
+	By("setup namespace for test")
+	namespace, err := generateNamespaceName()
+	Expect(err).NotTo(HaveOccurred())
+
+	runTest(mgr.GetClient(), namespace, ignoreOperationAnnotation)
+}
+
+func createAndStartManager(ignoreOperationAnnotation bool) (manager.Manager, error) {
+	mgrContext, mgrCancel := context.WithCancel(ctx)
+
+	framework.AddCleanupAction(func() {
+		defer func() {
+			By("stopping manager")
+			mgrCancel()
```

Comment:
Even though the cleanup action is being called and the metric server of test 1 should be shut down - when running test 2 the metric server address seems to be still in use.
```
error listening on :8080: listen tcp :8080: bind: address already in use
```
Possibly it does not shutdown quick enough.
Can we disable the server or why do we need it?

Reply:
How do you reproduce this? It's the first time I see it.

Reply:
Anyways, I'll set `MetricsBindAddress` to `0` in order to disable metrics serving. :)

Reply:
I think it happened during the debug session.
Thanks.